lib/ansible/modules/get_url.py

```diff
@@ -511,14 +511,17 @@ def main():
             os.remove(checksum_tmpsrc)
             checksum_map = []
             for line in lines:
-                parts = line.split(None, 1)
+                # Split by one whitespace to keep the leading type char ' ' (whitespace) for text and '*' for binary
+                parts = line.split(" ", 1)
                 if len(parts) == 2:
-                    checksum_map.append((parts[0], parts[1]))
+                    # Remove the leading type char and use basename
+                    checksum_map.append((parts[0], os.path.basename(parts[1][1:])))
```

Comment:
Should this strip the leading type char and './' to preserve potential paths? `basename` just gets the filename.

Reply:
good point.